user-interface/src/lib/components/combobox/ComboBox.tsx

```diff
@@ -302,24 +349,39 @@ function ComboBoxComponent(props: ComboBoxProps, ref: React.Ref<ComboBoxRef>) {
           }
         }
         if (index > 1 && index <= filteredOptions.length) {
-          const button = list?.children[index - 2].querySelector('button');
-          if (list && button) focusAndHandleScroll(ev, button);
-        } else if (index === 1) {
-          if (input) focusAndHandleScroll(ev, input);
+          const li = list?.children[index - 2];
+          if (list && li) {
+            focusAndHandleScroll(ev, li);
+            setCurrentListItem(li?.id);
+          }
+        } else {
+          let element;
+          if (selections.length && !multiSelect) {
+            element = singleSelectionPillRef.current;
+          } else if (input) {
+            element = input;
+          }
+          if (element) {
+            focusAndHandleScroll(ev, element);
+            setCurrentListItem(undefined);
+          }
           closeDropdown(true);
         }
         break;
       case 'Enter':
         if (!(ev.target as HTMLInputElement).classList.contains('combo-box-input')) {
-          handleDropdownItemSelection(option as ComboOption);
+          handleDropdownItemSelection(option as ComboOption, isSelected ?? false);
```

Comment:
`isSelected` is either `true` or falsy.
```suggestion
          handleDropdownItemSelection(option as ComboOption, isSelected);
```